ttnn/cpp/ttnn/operations/core/to_layout/to_layout_op.cpp

```diff
@@ -109,7 +110,9 @@ Tensor to_layout_impl(
 
     auto padded_output_shape = output_shape;
     for (auto index = output_shape.size() - 2; index < output_shape.size(); ++index) {
-        padded_output_shape[index] = ttnn::pad_to_multiple_of_tile_size(padded_output_shape[index]);
+        padded_output_shape[index] = (index == output_shape.size() - 2) ?
+            ttnn::pad_to_multiple_of_tile_size(padded_output_shape[index], tile.get_tile_shape()[0]) :
+            ttnn::pad_to_multiple_of_tile_size(padded_output_shape[index], tile.get_tile_shape()[1]);
```

Comment:
This still has method calls on both sides of the conditional.

Please do something like
```
padded_output_shape[index] = ttnn::pad_to_multiple_of_tile_size(padded_output_shape[index], 
(index == output_shape.size() - 2) ?
tile.get_tile_shape()[0]) : tile.get_tile_shape()[1]);
```